### DISCUSSION TITLE
Module based snapshots

### DISCUSSION THREAD

**Discussion by ernst86913** [directus/directus#16981] [Feature Requests]
To make snapshots work for module bundles which come with their own collections / fields / relations, it would be great to extend functionality of current snapshot feature to filter for collections which are relevant for a specific module - adding an array parameter to the cli calls.

With this functionality, a snapshot could be limited to the relevant data structures… and also applying a snapshot would only be limited to those data structures instead of applying it to holistic Schema.

Thinking a bit ahead (directus marketplace) it would actually be even better to think about a separate extension for this (including seeding of data) to make sure the setup of data / data structures (and deletion in case a module is removed) happen in a standardized way.

i know - data migrations functionality could be used for this… but in this case the amazing directus data studio becomes irrelevant as everything needs to be coded again…

any thoughts?

**Comment by rijkvanzanten** [comment]
Yes! We're actually looking into partial schema apply in https://github.com/directus/directus/pull/14713 as well 🙂

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
